### PR TITLE
update to show plots within Jupyter notebook in VSCode instead of browser

### LINKED ISSUE #34: Plots in VSCode Jupyter notebook opens in browser instead of below cell
Hi,

I'm trying to use the example plot script, however I end up with the plot opening up in the browser instead of below the cell in Jupyter notebook in VSCode.

``` Julia
using PlotlyLight

p = Plot(x = 1:20, y = cumsum(randn(20)), type="scatter", mode="lines+markers")  # Make plot

display(p)
```

The output of that is:
```
Process(`'C:\WINDOWS\system32\cmd.exe' /c start '%USERPROFILE%\.julia\scratchspaces\ec354790-cf28-43e8-bb59-b484409b7bad\CobWeb\index.html'`, ProcessRunning)
```

Running the same code in REPL creates the plot in a window withing VSCode which is how I would expect it to behave.

I'm running on Windows:
Julia v1.9.0
VSCode v1.82.3
PlotlyLight v0.7.4

### PR DESCRIPTION
**Pull request by Titas22**
resolves #34

- added `auto!()` container case for Jupyter running in VSCode to use IFrame without scrolling
- updated display function to show different output in this case

### CODE REVIEW

File: src/PlotlyLight.jl [hunk 1]
```diff
@@ -252,7 +255,13 @@ function page(o::Plot)
...
 end
 
-Base.display(::Cobweb.CobwebDisplay, o::Plot) = display(Cobweb.CobwebDisplay(), page(o))
+function Base.display(::Cobweb.CobwebDisplay, o::Plot) 
+    if (~isinteractive() && isdefined(Main, :VSCodeServer)) # VSCode Jupyter
+        display(HTML(repr("text/html", o)));
+    else
+        display(Cobweb.CobwebDisplay(), page(o))
+    end
+end
```
Comment (joshday) [inline src/PlotlyLight.jl:264]: I don't think this is the right approach.  You're using the `display(::CobwebDisplay, x)` method to possibly not use that display.

Comment (Titas22) [reply src/PlotlyLight.jl:264]: Would it make sense to do something like this then?
```
function Base.display(::Cobweb.CobwebDisplay, o::Plot) 
    if (~isinteractive() && isdefined(Main, :VSCodeServer)) # VSCode Jupyter
        display(Cobweb.CobwebDisplay(), HTML(repr("text/html", o)))
    else
        display(Cobweb.CobwebDisplay(), page(o))
    end
end
```

the 1st `HTML(repr("text/html", o))` option actually works in VSCode REPL as well but not in standalone Julia REPL.

Or do you have a better suggestion?

Comment (Titas22) [reply src/PlotlyLight.jl:267]: resolved by setting display_object to Cobweb.IFrame for when jupyter is running in vscode

**Comment by Titas22** [comment]
Hi @joshday, are you happy with this solution?

**Comment by joshday** [comment]
Sorry for my radio silence here!  I'm happy with your solution, with the caveats:

1. There's been a breaking change in Cobweb which will change a few display/show things here.
2.  Recently I've been working on a PlotlyLight rewrite of sorts (https://github.com/joshday/PlotlyX.jl) that attempts to fix/simplify some of the display issues.  This will be the second rewrite of sorts in so many months, but it really is tricky getting the display stuff right in all environments.  I think I'm on the right track this time.

**Comment by joshday** [comment]
Check out master or the new latest release!  It should eliminate the need for this PR since the display system is improved from before.

**Pull request closed without merging**